src/detect.h

```diff
@@ -1130,8 +1134,17 @@ typedef struct DetectEngineCtx_ {
     HashTable *non_pf_engine_names;
 
     const char *firewall_rule_file_exclusive;
+
+    /* user provided rate filter callbacks. */
+    SCDetectRateFilterFunc rate_filter_callback;
+
+    /* use provided data to be passed to rate_filter_callback. */
+    void *rate_filter_callback_arg;
```

Comment:
This could also be global, but I'm trying avoid that for the future library.

Reply:
Good by me. Just need to make sure that the setting of the field is done for reloads as well? Or is that the lib consumer responsibility?